debian/ubuntu-advantage-tools.postinst

```diff
@@ -318,6 +319,28 @@ enable_periodic_license_check() {
     fi
 }
 
+disable_new_timer_if_old_timer_already_disabled() {
+    # If the user has disabled the ua-messaging
+    # then we will assume that the user would also not want the
+    # ua-timer to be disabled as well. In tha case, we will
+    # disable the ua-timer here.
+    PREVIOUS_PKG_VER=$1
+
+    # We should only perform this check on UA version that have the
+    # ua-messaging.timer: 27.0 until 27.2. This will also guarantee
+    # that on 27.3 and forward, we will not run this logic.
+    if dpkg --compare-versions "$PREVIOUS_PKG_VER" ge-nl "27.0~"; then
+        if dpkg --compare-versions "$PREVIOUS_PKG_VER" lt-nl "27.3~"; then
+            if [ -x "/usr/bin/deb-systemd-helper" ]; then
+                if ! deb-systemd-helper --quiet was-enabled $OLD_MESSAGING_TIMER; then
+                    systemctl stop $UA_TIMER_NAME
+                    systemctl mask $UA_TIMER_NAME
```

Comment:
I was playing around with an alternative to masking just now, it looks like this
```
deb-systemd-helper enable 'ua-timer.timer' >/dev/null || true
systemctl stop $NEW_TIMER > /dev/null || true
systemctl disable $NEW_TIMER >/dev/null || true
```
But I haven't tested it thoroughly and its friday afternoon so I won't get back to it til monday :)

The theory behind it is that by calling `deb-systemd-helper enable` and then calling `systemctl disable`, then the `was-enabled` check later in the postinst script will return false.

Reply:
We should also look at what output is being generated from this section during install and make sure its not too confusing (my suggestion here might cause particularly confusing output)

Reply:
@lucasmoura Okay I've arrived at a suggestion. Here it is:
```suggestion
                    echo "$OLD_MESSAGING_TIMER was disabled. Disabling $UA_TIMER_NAME." >&2
                    deb-systemd-helper enable $UA_TIMER_NAME > /dev/null 2>&1 || true
                    systemctl stop $UA_TIMER_NAME > /dev/null 2>&1 || true
                    systemctl disable $UA_TIMER_NAME > /dev/null 2>&1 || true
```

Then when upgrading we get the following output:
```
ua-messaging.timer was disabled. Disabling ua-timer.timer.
Created symlink /etc/systemd/system/multi-user.target.wants/ua-license-check.path → /lib/systemd/system/ua-license-check.path.
ua-timer.timer is a disabled or a static unit, not starting it.
```

The nice thing about this is that the new timer ends up in the same state as the old timer (disabled) instead of a different state (masked).

The not nice thing about this is that the code is unintuitive and could be construed as "tricking the later debhelper snippet into thinking it was disabled". I think we should add a comment explaining what this is doing if we go this route.

Let's talk tradeoffs after standup

Reply:
LGTM. It would be nice to comment the approach, as suggested, since this may not be obvious, as @orndorffgrant stated.

Reply:
Sorry, I don't know if I understand @orndorffgrant's comment when he says:

> The theory behind it is that by calling `deb-systemd-helper enable` and then calling `systemctl disable`, then the `was-enabled` check later in the postinst script will return false.

I'm only seeing one call to `deb-systemd-helper was-enabled` in the code.

Reply:
Yes sorry for the poor explanation @sergiodj 

debhelper appends the following at the end of our postinst (among other things)
```bash
# Automatically added by dh_systemd_enable/11.1.6ubuntu2
if [ "$1" = "configure" ] || [ "$1" = "abort-upgrade" ] || [ "$1" = "abort-deconfigure" ] || [ "$1" = "abort-remove" ] ; then
        # This will only remove masks created by d-s-h on package removal.
        deb-systemd-helper unmask 'ua-timer.timer' >/dev/null || true

        # was-enabled defaults to true, so new installations run enable.
        if deb-systemd-helper --quiet was-enabled 'ua-timer.timer'; then
                # Enables the unit on first installation, creates new
                # symlinks on upgrades if the unit file has changed.
                deb-systemd-helper enable 'ua-timer.timer' >/dev/null || true
        else
                # Update the statefile to add new symlinks (if any), which need to be
                # cleaned up on purge. Also remove old symlinks.
                deb-systemd-helper update-state 'ua-timer.timer' >/dev/null || true
        fi
fi
# End automatically added section
```

Because we have this line in our `debian/rules`:
```
dh_systemd_enable -pubuntu-advantage-tools ua-license-check.timer
```

Reply:
Ah, of course, no need to apologize @orndorffgrant, I just thought you were referring to a call to `deb-systemd-helper` that got removed by mistake.  Indeed, I can see where you're coming from now, thanks.

Yeah, I agree the suggestion is a hack but I cannot think of a better approach right now.  I'm a strong +1 to properly document it on the postinst script, and I think we can proceed with this solution.